### DISCUSSION TITLE
Resume cleaning after "Vacuum: Return to dock"

### DISCUSSION THREAD

**Discussion by Hutch79** [Tasshack/dreame-vacuum#617] [Q&A] [answered]
Hey there 👋 

First of all, this integration is so great!
It's my first time using Home Assistant and it just works as expected. Love the work done here, thanks very much!

So to my question:
I have an automation which starts cleaning if nobody is home.
If somebody comes home, the vacuum currently stops and returns to the base station. I use the "Vacuum: Return to dock" for this.
If the cleaning was stopped, it'll start a new cleaning if he's alone again.

The problem here is, on a restart, dreame starts a new clean. Is it somehow possible to resume the cleaning sequence after returning to the dock?

**Comment by Tasshack** [comment, marked as answer]
If your device is the one with the self-wash base you can trigger self-clean for vacuum to return washing the mop pads and pause just after self-clean has been started. For continuing the cleaning task just start self-clean again and vacuum will continue cleaning task after self-clean has been completed. Otherwise it is not possible to resume a cleaning task after return to home has been triggered. 
I think X40 has an extended DnD feature to make the cleaning task pause when DnD has started and resume after it ended but even that feature is not useful for your case.

**Reply by Hutch79** [reply]
Hey @Tasshack 

Thanks for your answer!
I have a L10s Ultra with self-cleaning.

Your suggestion, with sending it to self-clean and pause/resume, works perfectly.
The only problem I'm currently facing is, how do I pause the self-cleaning?
I prepared an automation to pause it when self-cleaning starts, but I don't know which trigger to use.

**Reply by Tasshack** [reply]
You can use built-in entity services for setting any setting or calling an action.

In your case you need to press the Self-Clean button like this;
```
- service: input_button.press
  target:
    entity_id: button.{your vacuum name}_self_clean
```
https://www.home-assistant.io/integrations/input_button/#services

You can also use same button entity for resuming the cleaning task but the problem with this solution is the fact that your mop pads will be wet and dirty while vacuum is sitting at the dock so you may want to add a timeout for how long will vacuum wait at `Washing Paused` state.

**Reply by Hutch79** [reply]
Hey @Tasshack 
Sorry for the delayed response.

I managed so it does self clean, but I somehow need to pause self-cleaning, otherwise it would just resume cleaning after washing.
The "Vacuum Pause" action does work for that, and also works with resuming, but i don't know which trigger would be suitable to detect if the robot is currently in the doc and has started self wash.
